skrifa/src/outline/autohint/metrics.rs

```diff
@@ -218,40 +218,31 @@ pub(crate) struct Scale {
 }
 
 impl Scale {
-    /// Create initial scaling parameters from font size, units per em
-    /// and hinting mode.
-    pub fn new(size: f32, units_per_em: i32, mode: HintingMode) -> Self {
+    /// Create initial scaling parameters from metrics and hinting target.
+    pub fn new(size: f32, units_per_em: i32, target: Target, is_italic: bool) -> Self {
```

Comment:
is_italic is an odd argument here and does not feel in the right place. Can Skrifa determine on its own if the font is italic? What is supposed to be passed in here and what algorithm do you suggest for the client to determine this boolean - and if there is one, could that algorithm be run in Skrifa itself?

I think if there is really a client side decision needed on whether hinting should be activated for an italic font, then the client would need to adjust the hinting configuration / options for it?

Reply:
Definitely not a client side decision… we’ll handle this internally when the autohinting instance state lands. See https://github.com/googlefonts/fontations/blob/cfcd5975d4e3050184ddd80b08cb48fbfdda19fc/skrifa/src/outline/autohint/instance.rs#L77 in the preview branch.

Reply:
Good to hear this is not a client side setting. 

My concern was that we would expose a public `Scale` API with the odd boolean argument. But looking at this again, `Scale` is not a public Skrifa interface, is it? - If so, that addresses my concern. If possible, I would still prefer to turn the Boolean argument into an enum to make callsites clearer to read.

Reply:
Yes, this is internal code but fair point about the bare boolean. When I add the instance code, I’ll separate the hinting flags computation from scale setup and avoid the bool altogether.

Reply:
Reworked `Scale::new()` to take a `Style` parameter rather than a boolean

Reply:
Thanks for doing that, LGTM.